zerver/actions/users.py

```diff
@@ -498,7 +654,15 @@ def do_deactivate_user(
         # in only the user being deactivated.
         bot_profiles = get_active_bots_owned_by_user(user_profile)
         for profile in bot_profiles:
-            do_deactivate_user(profile, _cascade=False, acting_user=acting_user)
+            bot_group_setting_updates = check_group_permission_updates_for_deactivating_user(
```

Comment:
This will result in queries in a loop. It also happens in `delete_user` management command which allows deleting multiple users. Still working on updating this to use a bulk query.